packages/@aws-cdk/pipelines/lib/blueprint/synth-step.ts

```diff
@@ -0,0 +1,150 @@
+   *
+   * @default - No installation commands
+   */
+  readonly installCommands?: string[];
+
+  /**
+   * Environment variables to set
+   *
+   * @default - No environment variables
+   */
+  readonly env?: Record<string, string>;
+
+  /**
+   * FileSet with the CDK app source
+   *
+   * The files in the FileSet will be in the working directory when
+   * the script is executed.
+   *
+   * @default - No input specified
+   */
+  readonly input?: IFileSetProducer;
+
+  /**
+   * Additional FileSets to put in other directories
+   *
+   * Specifies a mapping from directory name to FileSets. During the
+   * script execution, the FileSets will be available in the directories
+   * indicated.
+   *
+   * The directory names may be relative. For example, you can put
+   * the main input and an additional input side-by-side with the
+   * following configuration:
+   *
+   * ```ts
+   * const script = new ScriptStep('MainScript', {
+   *   // ...
+   *   input: MyEngineSource.gitHub('org/source1'),
+   *   additionalInputs: {
+   *     '../siblingdir': MyEngineSource.gitHub('org/source2'),
+   *   }
+   * });
+   * ```
+   *
+   * @default - No additional inputs
+   */
+  readonly additionalInputs?: Record<string, IFileSetProducer>;
+
+  /**
+   * Directory that contains the Cloud Assembly output
+   *
+   * @default 'cdk.out'
+   */
+  readonly cloudAssemblyOutputDirectory?: string;
+
+  /**
+   * Additional directories that contain output FileSets.
+   *
+   * A map of output name to relative directory name (N.B.: this is
+   * reversed from `additionalInputs`).
+   *
+   * After running the script, the contents of the given directory
+   * will be exported as `FileSet`s. The `FileSet` objects can be
+   * retrieved by calling `additionalOutut(name)`.
+   *
+   * The following example exports the `cdk.out` directory as primary
+   * output, and the `integ` directory as an additional output
+   * with the name `IntegrationTests`:
+   *
+   * ```ts
+   * const script = new ScriptStep('MainScript', {
+   *   // ...
+   *   primaryOutputDirectory: 'cdk.out',
+   *   additionalOutputDirectories: {
+   *     IntegrationTests: 'integ',
+   *   }
+   * });
+   * ```
+   *
+   * @default - No additional outputs
+   */
+  readonly additionalOutputDirectories?: Record<string, string>;
+}
+
+/**
+ * A `ScriptStep` that knows about CDK app synthesis
+ *
+ * This class adds some defaults on top of a regular `ScriptStep`:
+ *
+ * - It knows that the default output directory of a synth
+ *   command is `cdk.out`.
+ * - It supports a `subdirectory` property, which you can use
+ *   if your CDK app lives somewhere other than the root of the
+ *   repository.
+ *
+ * This is a convenience class you do not have to use; you can also use a
+ * regular `ScriptStep` or `CodeBuildStep` instead. If you do, do not forget to
+ * pass `primaryOutputDirectory: 'cdk.out'`.
+ */
+export class SynthStep extends ScriptStep {
```

Comment:
Yeah, it's tough.

I feel it has *some* value in that we get to hide the `cdk.out` output (ugly internals of CDK I'd like to hide), as well as the notion that a CDK app might not live in the root of the repository.

I have a hard time with this one, but I dearly dearly like to hide `cdk.out`.

Reply:
I am wondering if perhaps we can add some magic that will use `cdk.out` as the default if a `ScriptStep` is used for `synth`... I also think hiding `cdk.out` is not that important and actually it might be good to surface it ([mechanical-sympathy](https://wa.aws.amazon.com/wellarchitected/2020-07-02T19-33-23/wat.concept.mechanical-sympathy.en.html)).